### PR TITLE
@mzikherman: Refactored error page

### PR DESCRIPTION
**Pull request by craigspaeth**
Solves: https://github.com/artsy/force/issues/1148

This refactors the error page to reduce dependencies so it can render any error. You may have seen `asset is not a function` or `Cannot read property 'CURRENT_USER' of undefined`. This is because our error page used the full main layout code which has things like the auth modal, logged in states, etc. which can't be rendered when errors occur higher up the middleware stack (e.g. being blocked by IP filter). I refactored the error page to inline styles, not need javascript, and wrapped the header in a `try` block so it won't choke on irrelevant errors.

If it fails to render the header the error page looks like this:

screen shot 2017-04-11 at 5 01 06 pm
screen shot 2017-04-11 at 5 01 16 pm

### CODE REVIEW

File: desktop/components/main_layout/footer/lower.jade [hunk 1]
```diff
@@ -1,14 +1,18 @@
 .mlf-lower: .main-layout-container
   nav#main-layout-footer-internal
-    a#main-layout-footer-logo.icon-logo( href='/' )
+    a#main-layout-footer-logo( href='/' )
+      include ../public/icons/logo.svg
```
Comment (craigspaeth) [inline desktop/components/main_layout/footer/lower.jade]: If the error is higher up the stack than [bucketAssets](https://github.com/artsy/force/blob/master/lib/setup.js#L182) the icon-font files won't load properly and you'll see square boxes. Another point for to inline SVG!

File: desktop/components/error_handler/index.jade [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+html
+  head
+    meta( name='viewport', content='width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no' )
+    style
+      include:stylus index.styl
+  body.error-handler-body
+
+    //- Try to render the full header, but if the error is high up the stack
+    //- then we likely won't have all of the locals needed to render this.
+    - try {
```
Comment (mzikherman) [inline desktop/components/error_handler/index.jade]: ooo fancy!

Comment (mzikherman) [inline desktop/components/error_handler/index.jade]: That's crazy you can do this in jade!

File: desktop/components/error_handler/index.styl [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    text-align center
+    width 100vw
+    margin 0
+    padding 0 20px
+    top calc(50% \- 70px)
+    left 50%
+    transform translate(-50%, -50%)
+    a
+      cursor pointer
+      -webkit-tap-highlight-color rgba(26, 26, 26, 0.301961)
```
Comment (mzikherman) [inline desktop/components/error_handler/index.styl]: lol what a specific number

File: lib/setup.js [hunk 4]
```diff
@@ -205,4 +205,12 @@ export default function (app) {
   // Ensure CurrentUser is set for Artsy Passport
   // TODO: Investigate race condition b/t reaction-force's use of AP
   artsyPassport.options.CurrentUser = CurrentUser
+
+  // Last but not least, add the error handler
```
Comment (mzikherman) [inline lib/setup.js]: 👍

**Comment by mzikherman** [comment]
LGTM!

**Pull request merged**